src/ashell/jerry-code.c

```diff
@@ -133,9 +151,10 @@ void javascript_stop()
 
     /* Cleanup engine */
     jerry_cleanup();
+    zjs_ipm_free_callbacks();
+    zjs_timers_free_timers();
 
-    /* Initialize engine */
-    jerry_init(JERRY_INIT_EMPTY);
+    restore_zjs_api();
```

Comment:
I understand the logic, but I am not sure restore_zjs_api should be called inside javascript_stop(), it seems it should be called after stop() and then before start() when you decided to run another app.